src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -654,46 +647,45 @@ private static Task UpdateMergeStatusAsync(IRemote darc, string prUrl, IReadOnly
                 SourceSha = sourceSha,
                 SourceRepo = sourceRepo,
                 Assets = assets,
-                IsCoherencyUpdate = false
+                IsCoherencyUpdate = false,
             };
 
             try
             {
-                if (pr != null)
+                if (pr == null)
                 {
-                    if (!canUpdate)
+                    string? prUrl = await CreatePullRequestAsync([updateParameter]);
+                    if (prUrl == null)
                     {
-                        _logger.LogInformation("Pull Request {url} cannot be updated at this moment", pr.Url);
-
-                        await _stateManager.AddOrUpdateStateAsync(
-                            PullRequestUpdate,
-                            new List<UpdateAssetsParameters> { updateParameter },
-                            (n, old) =>
-                            {
-                                old.Add(updateParameter);
-                                return old;
-                            });
-                        await _reminders.TryRegisterReminderAsync(
-                            PullRequestUpdate,
-                            [],
-                            TimeSpan.FromMinutes(5),
-                            TimeSpan.FromMinutes(5));
-                        return ActionResult.Create<object>(
-                            null,
-                            $"Current Pull request '{pr.Url}' cannot be updated, update queued.");
+                        return ActionResult.Create("Updates require no changes, no pull request created.");
                     }
 
-                    await UpdatePullRequestAsync(pr, [updateParameter]);
-                    return ActionResult.Create<object>(null, $"Pull Request '{pr.Url}' updated.");
+                    return ActionResult.Create($"Pull request '{prUrl}' created.");
                 }
 
-                string prUrl = await CreatePullRequestAsync([updateParameter]);
-                if (prUrl == null)
+                if (!canUpdate)
                 {
-                    return ActionResult.Create<object>(null, "Updates require no changes, no pull request created.");
+                    _logger.LogInformation("Pull Request {url} cannot be updated at this moment", pr.Url);
```

Comment:
Maybe we can remove this line, we're writing the same information a bit lower, on line 684

Reply:
Oh yeah, good catch. I think I can improve this method even more actually.